src/components/PlaylistPage/MobileComponents.js

```diff
@@ -28,7 +28,7 @@ const MobileComponents = React.createClass({
 
             <Collapse in={this.state.showFilter}>
               <div>
-                <LessonFilter filter={this.props.filter} onFilterCheck={this.props.onFilterCheck}/>
+                <LessonFilter {...this.props}/>
```

Comment:
Det at vi bruker `...this.props` alle steder vi kaller LessonFilter, tyder på at LessonFilter burde være wrappet i en container som henter state fra state-treet, og så bare la alle de andre komponentene rendre LessonFilterContainer.

Vi snakket jo om at ved å la en komponent rendre en Container i stedet for at bare "pages" får hente state, så blir ikke moder-komponenten "pure". Men etter å ha tenkt litt, så er det vel bedre å tenke sånn at Moderkomponent + underkomponent fortsatt er "pure", for de får begge state fra state-treet. Altså at de ikke er pure relativ til propsene som kommer inn til moderkomponenten, men relativ til en gitt state i state-treet.

Uansett, pure eller ikke pure, vi har ikke stort med logikk i komponentene (bare render-logikk), og vi skriver jo ikke tester for dem (ihvertfall ikke ennå). Og jeg tror det blir mindre komplekst å ha store komponenter som LessonFilter som egne containere.

Kom gjerne med motargumenter, @kristandre ! Hvis du er enig med meg, kan du lage en issue på det da?

Jeg merger denne endringen uansett.